src/tmpfiles/tmpfiles.c

```
@@ -2598,23 +2595,20 @@ static int glob_item_recursively(
                 Item *i,
                 fdaction_t action) {
 
-        _cleanup_globfree_ glob_t g = {
-                .gl_opendir = (void *(*)(const char *)) opendir_nomod,
-        };
+        _cleanup_strv_free_ char **paths = NULL;
         int r;
 
         assert(c);
         assert(i);
         assert(action);
 
-        r = safe_glob(i->path, GLOB_NOSORT|GLOB_BRACE, &g);
+        r = safe_glob_full(i->path, GLOB_NOSORT|GLOB_BRACE, opendir_nomod, &paths);
         if (r == -ENOENT)
                 return 0;
         if (r < 0)
                 return log_error_errno(r, "Failed to glob '%s': %m", i->path);
 
-        r = 0;
```

Why is this dropped?

---

safe_glob() returns 0 or negative errno. Hence, `r` is already 0 here.
Moreover, the caller of this function does not care if this function returns 0 or any positive value, hence it is not necessary to initialize `r` with 0 here.

---

Yeah, but I'd rather not rely on that here. We typically initialize r before any use of RET_GATHER explicitly

---

I do not think so? It is necessary only when some earlier error will be ignored.

---

I agree with @YHNdnzj on this, I think it's easier to grok if this is reset to zero here. Who knows, maybe safe_glob() returns > 0 one day, and then we propagate something that we probably shouldn't.

---

(or to say this differently: it's much easier to follow the code too I think if the loop below starts with clear initialization, instead of making the reader understand that any state inherited from the earlier initialization of `r` is not relevant)

---

OK. The removal is dropped.